astropy/io/ascii/tests/test_read.py

```diff
@@ -158,6 +161,50 @@ def test_read_all_files_via_table(fast_reader):
                 assert_equal(len(table[colname]), testfile['nrows'])
 
 
+@pytest.mark.parametrize('fast_reader', [True, False, 'force'])
+def test_read_all_files_with_encoding(fast_reader, tmpdir):
+    from ....table import Column
+    formats = {v: k for k, v in core.FORMAT_CLASSES.items()}
+
+    for testfile in get_testfiles():
+        if testfile.get('skip'):
+            print('\n\n******** SKIPPING %s' % testfile['name'])
+            continue
+
+        tmpfile = str(tmpdir.join(os.path.basename(testfile['name'])))
+
+        table = ascii.read(testfile['name'], **testfile['opts'])
+        if table.columns[0].dtype.kind == 's':
+            name = u'à' if not six.PY2 else 'alpha'
+            col = Column(name=name, data=[six.u(x) for x in table.columns[0]])
+            table.add_column(col, 0)
+            table[0][0] = u"àéö"
```

Comment:
Later on this table value should be tested.  After this block you could define a variable `table00 = table[0][0]` and then test.